### ISSUE TITLE
How to create a Simple Consumer (detached from a group)?

### ISSUE THREAD

**Issue by smoya** [confluentinc/confluent-kafka-go#534]
Description
===========

I'm wondering how can I create a Simple Consumer, detached from a Consumer Group. 
I tried by skipping the `Subscribe` in favor of `Assign`, and assigning a random consumer group id (Because this library requires it, but I expect it does nothing with it?), however checking the logs I still see group-related things, E.g. trying to use a group which our policy doesn't allow:

```
FindCoordinator response error: Not authorized to access group: Group authorization failed.
```

Checklist
=========
Please provide the following information:

 - [x] confluent-kafka-go and librdkafka version (`LibraryVersion()`): `v1.4.2` (both)
 - [x] Apache Kafka broker version: `2.3.1`
 - [x] Operating system: Ubuntu 16.04 and macOS catalina.

**Comment by edenhill** [comment]
It will still use FindCoordinator to find the group coordinator, since this is needed for commits, and there is no way to opt out of that lookup at the present time.
So unfortunately there is currently no way to run a simple consumer if you don't have FindCoordinator permissions in the cluster.

**Comment by smoya** [comment]
Thanks for your super fast answer @edenhill. I'm gonna add more details about my use case: 

Different consumers need to consume from the same partition at the same time. The number of consumers is unknown because those are created "almost" (let's simplify) on each request my server receives, so I can't have a predefined fleet of them.
Using consumer groups it's not viable since I would need to create one consumer group per consumer in order to let a partition to be consumed concurrently.

Is that possible? I was thinking that if the share the same consumer group id it would work if Subscribe is not call ever. Not sure now :\

**Comment by smoya** [comment]
Let me rephrase, in case the consumers can access the group coordinator, would that work? By work I mean no partition assignment by any group coordinator nor rebalance will happen.

**Comment by edenhill** [comment]
Your design works ss long as you don't call subscribe(), just assign(),
given that FindCoord works, and you should disable auto commits since they
won't make sense with multiple readers for the same partition.

Den tors 24 sep. 2020 13:58Sergio Moya <notifications@github.com> skrev:

**Comment by smoya** [comment]
Awesome, thanks @edenhill . I think it is worth to document that in the readme or as an example. Would you mind keeping this issue opened? i would most probably work on providing that next week. Otherwise I can open another one.

**Comment by smoya** [comment]
I confirm the solution works as expected 🎉
